### PR TITLE
Added Windows compatibility and various runtime error fixes

### PR DESCRIPTION
**Pull request by pryon**
Added Windows compatibility by:

- wrapping *nix-exclusive functions into macros and adding Windows-compatible substitutes
- passing the argc, argv values to real_main and dismissing them in the function body
- adding macro to use strcpy_s in place of strcpy when compiling with MSVC

Fixed bugs:

- Allegro's primitives addon isn't initialized
- sometimes already free'd memory is being accessed when adding score to total
- when linking, old versions of GCC require that the object files come before the libraries

Builds tested on:

- Windows 7 64-bit with Visual Studio 2015 (Allegro 5.2.2)
- Ubuntu 15.04 32-bit with GCC 4.9.2 (Allegro 5.2.2)

### CODE REVIEW

File: src/main.c [hunk 1]
```diff
@@ -8,22 +8,35 @@
...
 
-int real_main() {
-    srand48(time(NULL));
+int real_main(int argc, char **argv) {
+
+	/* handling unused parameters without warnings from any compiler */
+	(void)argc;
+	(void)argv;
+
+	SRAND48();
```
Comment (lclarkmichalek) [inline src/main.c]: You indentation in this file is inconsistent with the rest of the file

**Comment by lclarkmichalek** [comment]
Hey! Looks good, only one comment. I noticed you're using tabs for indentation, would it be possible to change that to 4 spaces. The use after free bug you fixed was really nasty, nice catch.

Out of interest, what inspired you to fix these problems? I wrote this as a piece of coursework when in university, and didn't expect anyone to ever look at it again :+1:

**Comment by pryon** [comment]
Hi! Yeah, I didn't notice the problem with the indentation until I committed, but eventually forgot about it. Thanks for pointing it out!

As for inspiration, I myself did read the Head First C book and attempted to implement the same game using the book as a guide. I'm pretty much finished and already made the repo for it, but don't want to upload the files until I get it working properly both on Linux and Windows. I still have a few bugs to fix that are exclusive to Linux but other than that it's almost done. It will be uploaded to [here](https://github.com/pryon/Blasteroids) when I get around to it.

As for finding your repo, I was just googling for "head first C asteroids" to check out how others implemented some features that I had trouble with (namely collision detection). I found your code easily, so I decided to fork it and mess around with it in my Windows dev environment. This is pretty much how I found the bugs, and why I ported it to Windows. :)

**Comment by lclarkmichalek** [comment]
Awesome. Well, I think this repo still has a bug with the collision detection (the rotation matrix of the asteroid isn't applied during collision detection maybe), but I'm glad you found it useful.

**Pull request merged**